app/lib/atomic_lti/paging_helper.rb

```diff
@@ -0,0 +1,40 @@
+module AtomicLti
+  module PagingHelper
+    MAX_PAGES = 200
+
+    def self.response_link_urls(response, *rels)
+      links = response.headers["link"]&.split(",") || []
+      urls = {}
+      rels.each do |rel|
+        urls[rel] = link_url(links, rel)
+      end
+      urls.values_at(*rels)
+    end
+
+    def self.link_url(links, rel)
+      matching_link = links.detect { |link| link.include?("rel=\"#{rel}\"") }
+
+      return unless matching_link
+
+      matching_link.split(";")[0].gsub(/[<>\s]/, "")
+    end
+
+    def self.paginate_request(response)
+      next_url, = response_link_urls(response, "next")
+      response = yield response, next_url
```

Comment:
This might be cleaner as a do/while (or loop do) so you don't need two separate yields.